website/src/docs/migration-guides.md

```diff
@@ -8,7 +8,127 @@ category: "Docs"
 
 These cover all the major Uppy versions and how to migrate to them.
 
-## Migrate to 2.0.0
+## Migrate from Uppy 2.x to 3.x
+
+### Uppy is pure ESM
+
+Following the footsteps of many packages, we now only ship Uppy core and its plugins as
+[ECMAScript Modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) (ESM).
+On Uppy 2.x, we were shipping CommonJS.
+
+If are already using ESM yourself, or are using the CDN builds, nothing changes for you!
+
+If you are using CommonJS, you might need to add some tooling for everything to work, or you might
+want to refactor your codebase to ESM – refer to
+the [Pure ESM package](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c)
+gist for added information and help on how to do that.
+
+### Robodog is deprecated
+
+Uppy is flexible and extensible through plugins. But the integration code could sometimes be daunting. This is what brought [Robodog](/docs/robodog/) to life. An alternative with the same features, but with a more ergonomic and minimal API.
+
+But, it didn’t come with its own set of new problems:
+
+* It tries to do the exact same, but it looks like a different product
+* It’s confusing for users whether they want to use Robodog or Uppy directly.
+* Robodog is more ergonomic because it’s limited. When you hit such a limit, you need to refactor everything to Uppy with plugins.
+* Documentation is scarce, and the trade offs are unclear
+* It’s not marketed, you need to stumble on it in the docs.
+* Extra maintenance burden
+* Extra hosting effort and costs
+
+This has now led us to deprecating Robodog and embrace Uppy for its strong suits; modularity and flexibility. At the same time, we also introduced something to take away some repetitive integration code: [`@uppy/remote-sources`](/docs/remote-sources).
+
+To migrate from Robodog to Uppy plugins, go through the [list of plugins](/docs/plugins/) and install and use the ones you need.
```

Comment:
It'd be really nice to have a code example showing how one can replicate Robodog, showing how to migrate.

Reply:
That's what we don't want, I'd say. We actually discourage usage of the CDN bundle, because it includes everything. The same happens in Robodog, it will almost always include more than you need. It's better advice to pick only what you need from the plugins list, rather than recreate it.

I could point to the codesandbox of our dashboard setup for a quick copy paste reference.

Reply:
That's not my point, I'm saying we need to show what's the migration path for folks who are using Robodog today: something that shows how each Robodog feature can be replicated without it.